src/AlphaHomoraV2/mappings/HomoraBank.ts

```diff
@@ -0,0 +1,499 @@
+import { Address, BigInt, Bytes, BigDecimal, ByteArray, log } from "@graphprotocol/graph-ts";
+import {
+  PutCollateral,
+  TakeCollateral,
+  Borrow,
+  Repay,
+  Liquidate,
+  AlphaHomoraV2HomoraBank,
+} from "../../../generated/AlphaHomoraV2HomoraBank/AlphaHomoraV2HomoraBank";
+import { AlphaHomoraV2ERC20 } from "../../../generated/AlphaHomoraV2HomoraBank/AlphaHomoraV2ERC20";
+import { AlphaHomoraV2LpToken } from "../../../generated/AlphaHomoraV2HomoraBank/AlphaHomoraV2LpToken";
+import { AlphaHomoraV2CyToken } from "../../../generated/AlphaHomoraV2HomoraBank/AlphaHomoraV2CyToken";
+import { AlphaHomoraV2MasterChef } from "../../../generated/AlphaHomoraV2HomoraBank/AlphaHomoraV2MasterChef";
+import { AlphaHomoraV2MasterChefPNG } from "../../../generated/AlphaHomoraV2HomoraBank/AlphaHomoraV2MasterChefPNG";
+import { AlphaHomoraV2BoostedMasterChef } from "../../../generated/AlphaHomoraV2HomoraBank/AlphaHomoraV2BoostedMasterChef";
+import { AlphaHomoraV2BonusTokenRewarder } from "../../../generated/AlphaHomoraV2HomoraBank/AlphaHomoraV2BonusTokenRewarder";
+import { AlphaHomoraV2WMasterChef as wChefContract } from "../../../generated/AlphaHomoraV2HomoraBank/AlphaHomoraV2WMasterChef";
+import {
+  AlphaHomoraV2WMasterChef,
+  AlphaHomoraV2FarmData,
+  AlphaHomoraV2LpPair,
+  AlphaHomoraV2LendData,
+  AlphaHomoraV2Token,
+  AlphaHomoraV2CreamToken,
+} from "../../../generated/schema";
+import { convertBINumToDesiredDecimals } from "../../utils/converters";
+import { ZERO_ADDRESS, HOMORA_BANK_ADDRESS } from "../../utils/constants";
+
+/*
+handle put/take collateral on homora bank contract: event.params.id encoded from pool id and joePerShare at the time of the minting of the erc1155 token representing a position
+  - can use event.params.id w homora bank function 'decodeId' to get the pool id, as well as the underlying LP token from any of the wMasterChefContracts
+
+*/
+export function getOrCreateToken(tokenAddress: string): AlphaHomoraV2Token | null {
+  let tokenEntity = AlphaHomoraV2Token.load(tokenAddress);
+  if (!tokenEntity) {
+    tokenEntity = new AlphaHomoraV2Token(tokenAddress);
+    let contract = AlphaHomoraV2ERC20.bind(Address.fromString(tokenAddress.toString()));
+    let decimals = 18; // fallback
+    let decimalsResult = contract.try_decimals();
+    if (decimalsResult.reverted) {
+      log.warning("decimals() reverted", []);
+    } else {
+      decimals = decimalsResult.value;
+    }
+    decimals = BigInt.fromI32(decimals).toI32();
```

Comment:
Why is this needed? What does it do?

Reply:
accidentally left line 45 in, is not needed.